src/rust/ensogl/lib/core/src/display/object/class.rs

```diff
@@ -342,22 +342,26 @@ impl<Host> Model<Host> {
 
     /// Hide all removed children and show this display object if it was attached to a new parent.
     fn update_visibility(&self, host:&Host, parent_scene_layers:&[LayerId]) {
-        self.take_removed_children_and_hide_orphans(host);
+        self.take_removed_children_and_update_their_visibility(host);
         let parent_changed = self.dirty.parent.check();
         if parent_changed && !self.is_orphan() {
             self.set_vis_true(host,parent_scene_layers)
         }
     }
 
-    fn take_removed_children_and_hide_orphans(&self, host:&Host) {
+    fn take_removed_children_and_update_their_visibility(&self, host:&Host) {
         if self.dirty.removed_children.check_all() {
             debug!(self.logger, "Updating removed children.", || {
                 for child in self.dirty.removed_children.take().into_iter() {
                     if let Some(child) = child.upgrade() {
-                        if child.is_orphan() {
+                        if !child.has_visible_parent() {
                             child.set_vis_false(host);
```

Comment:
Yes, it may happen. But if the visible parent got hidden, then it will call set_vis_false, and thus the moved child will be also hidden.

Reply:
So this is bad. `set_vis_false` sohuld be guaranteed to be called only once and never be called on a thing already hidden. Otherwise it can break some rendering assumptions. The same with showing things.

Reply:
but set_vis_false checks if the instance is visible. I will add a test to make sure that on_hide callback is not called redundantly.